packages/react-utils/src/helpers/dataLoaders.ts

```diff
@@ -33,6 +37,86 @@ export class OpenSRPService<T extends object = Dictionary> extends GenericOpenSR
   }
 }
 
+/** A generic FHIR service class
+ *
+ * Sample usage:
+ * -------------
+ * const serve = new FHIRServiceClass('<base url>', '<resource type>');
+ *
+ * **To list all entries of a resource (GET request)**: serve.list()
+ *
+ * **To get one resource record**: service.read('<object id>')
+ *
+ * **To create a new resource**: service.create(payload)
+ *
+ * **To update a resource record**: service.update(payload)
+ */
+export class FHIRServiceClass<T = fhirclient.FHIR.Resource> {
+  public accessTokenOrCallBack: GetAccessTokenType | string;
+  public baseURL: string;
+  public resourceType: string;
+
+  /**
+   * Constructor method
+   *
+   * @param {string} baseURL - the base FHIR URL
+   * @param {string} resourceType - FHIR resource type string
+   */
+  constructor(baseURL: string, resourceType: string) {
+    this.accessTokenOrCallBack = handleSessionOrTokenExpiry;
+    this.baseURL = baseURL;
+    this.resourceType = resourceType;
```

Comment:
do we really need to specify the resource type ? if we really need it cant use use pick it from payload being sent or used ?

Reply:
we need it when making a GET request